### DISCUSSION TITLE
How to get theme primary color hex value

### DISCUSSION THREAD

**Discussion by rexkenley** [microsoft/fluentui#30260] [Q&A]
I am using a component from another ui library. I want to use Fluent UI theme provider primary color as the "primary color" of that component.

How do I do that and can I get the hex value?

**Comment by pradeept95** [comment]
Hi @rexkenley,
I believe you can achieve this by using tokens (or css variables if you using .css or .scss, etc. file) to style your custom or third party components. 

Here is the example that uses the primary/brand color scheme.
https://codesandbox.io/p/sandbox/competent-bas-tcclqd?file=%2Fsrc%2Fexample.tsx%3A26%2C11

This might not exactly answer your question but if you provide the more detail, there must be some solution. Please provide few more details like how and where you traying to use primary/brand color.

![image](https://github.com/microsoft/fluentui/assets/28040299/a7839727-8da0-474f-8d7f-46c1f783c347)

**Reply by rexkenley** [reply]
@pradeept95 

Thanks for you reply. I did try the token, but like your screenshot it returns a name. I need the color hex value.

https://mui.com/material-ui/guides/creating-themed-components/#introduction
![image](https://github.com/microsoft/fluentui/assets/10548251/cc4b1621-670f-4692-9dcd-02b1e6ea2c28)

**Reply by pradeept95** [reply]
@rexkenley ,
Oh I see, if you need hex color code, you can directly access it from the theme object like webLightTheme for instance. You can use the same theme object that you're using in you FluentProvider.

I have updated my example to get the hex value. Hope this will help!

https://codesandbox.io/p/sandbox/competent-bas-tcclqd?file=%2Fsrc%2Fexample.tsx%3A37%2C23

![image](https://github.com/microsoft/fluentui/assets/28040299/f8418661-662f-4190-b6ca-63cc68dda02e)

![image](https://github.com/microsoft/fluentui/assets/28040299/93910fbb-dc85-414e-93c8-687e62634d33)

![image](https://github.com/microsoft/fluentui/assets/28040299/997794ac-3c5b-474a-a405-9cf0c0370d02)

**Reply by rexkenley** [reply]
But what if I am using a custom theme? Does that mean I have to pass the theme via props?

Is there a way to get theme from the parent without the use of props?

**Reply by tombohub** [reply]
@pradeept95 thank you, this is good

**Reply by tombohub** [reply]
@rexkenley if you are using custom theme following these [guide](https://react.fluentui.dev/?path=/docs/concepts-developer-theming--page) then import that theme in component where you need to use the token and do `const color = customTheme.colorBrandBackground`

**Comment by rexkenley** [comment]
https://github.com/microsoft/fluentui/discussions/31171
